### PR TITLE
PR to merge master branch into PG15 branch

### PR DESCRIPTION
**Pull request by jrgemignani**
**PLEASE DO NOT MERGE** until thoroughly reviewed. Additionally, this merge needs to be appropriately squashed before merging.

PR for moving the PG15 branch up to the current master branch prior to the release.

It contains all of the latest work in the master branch, which is currently at PostgreSQL version 15.

### CODE REVIEW

**Review by rafsun42 — APPROVED** [review]
Looks good.

**Review by MuhammadTahaNaveed — APPROVED** [review]
Looks good!

File: RELEASE [hunk 1]
```diff
@@ -19,7 +19,7 @@ Release Notes for Apache AGE release 0.0.0 for PG 15
 
 Apache AGE 0.0.0 - Release Notes
 
-NOTE: This is an initial release of PG14. There are no upgrade
+NOTE: This is an initial release of PG15. There are no upgrade
       scripts to this release.
 
 **************************************************************
```
[the remark this answers is not in the document]
Comment (jrgemignani) [reply RELEASE:25]: Not really. It is just a template that will be adjusted prior to release.

**Review by dehowef — APPROVED** [review]
Looks good.

**Pull request merged**